app/src/main/java/com/example/pam_lab1/DestinationCreateActivity.kt

```diff
@@ -0,0 +1,148 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_destiny_create)
+
+//        setSupportActionBar(toolbar)
+        val context = this
+
+        // Show the Up button in the action bar.
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        btn_add.setOnClickListener {
+            val newDestination = Destination()
+            newDestination.course = et_course.text.toString()
+            newDestination.description = et_description.text.toString()
+            newDestination.due = et_due.text.toString()
+            newDestination.subject = et_subject.text.toString()
+
+            val destinationService = ServiceBuilder.buildService(DestinationService::class.java)
+            val requestCall = destinationService.addDestination(newDestination)
+
+            requestCall.enqueue(object: Callback<Destination> {
+
+                override fun onResponse(call: Call<Destination>, response: Response<Destination>) {
+                    if (response.isSuccessful) {
+                        finish() // Move back to DestinationListActivity
+                      //  var newlyCreatedDestination = response.body() // Use it or ignore it
+                        Toast.makeText(context, "Successfully Added", Toast.LENGTH_SHORT).show()
+                    } else {
+                        Toast.makeText(context, "Failed to add item", Toast.LENGTH_SHORT).show()
+                    }
+                }
+
+                override fun onFailure(call: Call<Destination>, t: Throwable) {
+                    Toast.makeText(context, "Failed to add item", Toast.LENGTH_SHORT).show()
+                }
+            })
+        }
+        gestureDetector = GestureDetector(this,this)
+    }
+
+    override fun onTouchEvent(event: MotionEvent?): Boolean {
+
+        gestureDetector.onTouchEvent(event)
+        when (event?.action){
+
+            0->{
+                x1 = event.x
+                y1 = event.y
+            }
+            1->{
+                x2 = event.x
+                y2 = event.y
+
+                val valueX:Float = x2 - x1
+                val valueY:Float= y2 - y1
+
+                if(abs(valueX) > MainActivity.MIN_DISTANCE){
+                    if(x2>x1){
+                        this.finish()
+                       // Toast.makeText(this,"Right Swipe",Toast.LENGTH_SHORT).show()
+                    }
+                    else{
+
+                        //Toast.makeText(this,"Left Swipe",Toast.LENGTH_SHORT).show()
+                    }
+                }
+                else if(abs(valueY) > MainActivity.MIN_DISTANCE){
+                    if(y2>y1){
+                        Toast.makeText(this,"Bottom Swipe",Toast.LENGTH_SHORT).show()
+                    }
+                    else{
+                        Toast.makeText(this,"Top Swipe",Toast.LENGTH_SHORT).show()
+                    }
+                }
+            }
+
+        }
```

Comment:
Why do yo need this code? What functionalities you wanted to build? Have you checked if there is some native android component which is doing same?
I have seen that you removed it in the next PR. If you don't need just remove it